src/iuse.cpp

```diff
@@ -8090,22 +8090,22 @@ int iuse::washclothes( player *p, item *it, bool, const tripoint& )
         return 0;
     }
     
-    const inventory &crafting_inv = p->crafting_inventory();
-    if( !crafting_inv.has_charges( "water", 40 ) && !crafting_inv.has_charges( "water_clean", 40 ) ) {
-        p->add_msg_if_player( _( "You need a large amount of fresh water to use this." ) );
-        return 0;
-    }
-    
     const int pos = g->inv_for_flag( "FILTHY", _( "Wash what?" ) );
     item &mod = p->i_at( pos );
     if( pos == INT_MIN ) {
         p->add_msg_if_player( m_info, _( "Never mind." ) );
         return 0;
     }
-
+    
+    const inventory &crafting_inv = p->crafting_inventory();
+    if( !crafting_inv.has_charges( "water", 3 * mod.volume() ) && !crafting_inv.has_charges( "water_clean", 3 * mod.volume() ) ) {
```

Comment:
`3 * mod.volume()` occurs four times. I think a constant should be used here:
`const auto required_water_charges  = 3 * mod.volume()`.